CONTRIBUTING.md

```diff
@@ -117,22 +69,70 @@ repo root, run:
 ```sh
 mkdir build
 cd build
-cmake -Duse_default_uuid=ON ..
+cmake ..
 cmake --build .
 ```
+
+#### CMake build options
+The following CMake options are available for adding/removing project features.
+
+<table>
+<tr>
+<td>Option</td>
+<td>Description</td>
+<td>Default Value</td>
+</tr>
+<tr>
+<td>BUILD_TESTING</td>
+<td>Generates Unit Test for compilation. CMake will automatically download and build g-test.<br>After Compiling, use `ctest` to run Unit Test.</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>BUILD_STORAGE_SAMPLES</td>
+<td>Build Azure Storage clients sample application.</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>RUN_LONG_UNIT_TESTS</td>
+<td>Enables the special unit tests which takes more than 3 minutes to run. THis tests are for some specific features like the connection pool for curl transport adapter.</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>WARNINGS_AS_ERRORS</td>
```

Comment:
Why do we have this option? And if we really need it, the default should be `ON`.

Reply:
Default is ON.
That's the same opt we have in C that would make compilation fail if there's any warning